apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -908,4 +957,115 @@ public void run() {
+         Collection<String> objectManifest = objInfo.get("X-Static-Large-Object");
+         return String.valueOf(objectManifest.toArray()[0]);
+      }
+      return null;
+   }
+   
+   
+   /**
+    * @param containerAndPrefix
+    */
+   private void removeObjectsWithPrefix(String containerAndPrefix) {
+      String[] parts = splitContainerAndKey(containerAndPrefix);
+
+      String container = parts[0];
+      String prefix = parts[1];
+
+      removeObjectsWithPrefix(container, prefix);
+   }
+
+    
+   /**
+    * @param containerAndKey
+    * @return array of size 2 with container and objectPrefix
+    */
+   private String[] splitContainerAndKey(String containerAndKey) {
+      String[] parts = containerAndKey.split("/", 2);
+      checkArgument(parts.length == 2, "No / separator found in \"%s\"", containerAndKey);
+      return parts;
+   }
+    
+   /**
+    * Retrieves a list of existing storage
+    * @param container
+    * @param options
+    * @return maximum of 10,000 object names per request.
+    */
+   private PageSet<? extends StorageMetadata> list(final String container,
+         org.jclouds.openstack.swift.v1.options.ListContainerOptions options) {
+      ObjectApi objectApi = api.getObjectApi(regionId, container);
+      ObjectList objects = objectApi.list(options);
+      if (objects == null) {
+         return new PageSetImpl<StorageMetadata>(ImmutableList.<StorageMetadata> of(), null);
+      } else {
+         List<? extends StorageMetadata> list = transform(objects, toBlobMetadata(container));
+         int limit = 10000;
+         String marker = null;
+         if (!list.isEmpty() && list.size() == limit) {
+            marker = list.get(limit - 1).getName();
+         }
+         list = transform(list, new Function<StorageMetadata, StorageMetadata>() {
+            public StorageMetadata apply(StorageMetadata input) {
+               if (input.getType() != StorageType.BLOB) {
+                  return input;
+               }
+               return blobMetadata(container, input.getName());
+            }
+         });
+         return new PageSetImpl<StorageMetadata>(list, marker);
+      }
+   }
+    
+   /**
+    * Removes object segments
+    * @param container
+    * @param prefix
+    * 
+    */
+   private void removeObjectsWithPrefix(String container, String prefix) {
+      String nextMarker = null;
+      ObjectApi objectApi = api.getObjectApi(regionId, container);
+      do {
+         org.jclouds.openstack.swift.v1.options.ListContainerOptions listContainerOptions = prefix(prefix);
+         if (nextMarker != null) {
+            listContainerOptions = listContainerOptions.marker(nextMarker);
+         }
+         PageSet<? extends StorageMetadata> chunks = list(container, listContainerOptions);
+         Iterator<? extends StorageMetadata> iter = chunks.iterator();
+
+         while (iter.hasNext()) {
+            objectApi.delete(iter.next().getName());
```

Comment:
Yes.  DLO generally is poorly thought and and supported.  Again, scope this commit down to just the provider changes and submit the portable abstraction changes as a separate pull request.